tools/regression_test.py

```diff
@@ -0,0 +1,1055 @@
+    logger.info(f'Processing regression test in {test_type} mode.')
+
+    backend_file_info = {
+        'onnxruntime': 'end2end.onnx',
+        'tensorrt': 'end2end.engine',
+        'openvino': 'end2end.xml',
+        'ncnn': ['end2end.param', 'end2end.bin'],
+        'pplnn': ['end2end.onnx', 'end2end.json'],
+        'torchscript': 'end2end.pt'
+    }
+
+    backend_list = args.backends
+    if backend_list == ['all']:
+        backend_list = [
+            'onnxruntime', 'tensorrt', 'openvino', 'ncnn', 'pplnn',
+            'torchscript'
+        ]
+    assert isinstance(backend_list, list)
+    logger.info(f'Regression test backend list = {backend_list}')
+
+    work_dir = Path(args.work_dir)
+    work_dir.mkdir(parents=True, exist_ok=True)
+
+    log_path = work_dir.joinpath('regression_test.log').absolute()
+    if log_path.exists():
+        # clear the log file
+        with open(log_path, 'w') as f_log:
+            f_log.write('')
+    logger.info(f'Saving log in {str(log_path.absolute().resolve())}')
+
+    for deploy_yaml in args.deploy_yml:
+
+        if not Path(deploy_yaml).exists():
+            raise FileNotFoundError(f'deploy_yaml {deploy_yaml} not found, '
+                                    'please check !')
+
+        with open(deploy_yaml) as f:
+            yaml_info = yaml.load(f, Loader=yaml.FullLoader)
+
+        report_save_path = \
+            work_dir.joinpath(Path(deploy_yaml).stem + '_report.xlsx')
+        report_txt_path = report_save_path.with_suffix('.txt')
+
+        report_dict = {
+            'model_name': [],
+            'model_config': [],
+            'task_name': [],
+            'model_checkpoint_name': [],
+            'dataset': [],
+            'backend_name': [],
+            'deploy_config': [],
+            'static_or_dynamic': [],
+            'precision_type': [],
+            'conversion_result': [],
+            'fps': []
+        }
+
+        global_info = yaml_info.get('globals')
+        for metric_name in global_info.get('metric_info', {}):
+            report_dict.update({metric_name: []})
+        metric_info = global_info.get('metric_info', {})
+        report_dict.update({'test_pass': []})
+
+        with open(report_txt_path, 'w') as f_report:
+            title_str = ''
+            for key in report_dict:
+                title_str += f'{key},'
+            title_str = title_str[:-1] + '\n'
+            f_report.write(title_str)  # clear the report tmp file
+
+        models_info = yaml_info.get('models')
+        for models in models_info:
+            if 'model_configs' not in models:
+                logger.warning('Can not find field "model_configs", '
+                               f'skipping {models.get("name")}...')
+                continue
+
+            model_metafile_info, checkpoint_save_dir, codebase_dir = \
+                get_model_metafile_info(global_info, models, logger)
+            for model_config in model_metafile_info:
```

Comment:
For now the model config in each model of test yaml is 1, but I don't know in the furture it will be 2 or more for each model. So I designed this to prevent multiple model config file under one model in the future. 😁